devops/scripts/benchmarks/benches/gromacs.py

```diff
@@ -0,0 +1,223 @@
+# Copyright (C) 2025 Intel Corporation
+# Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+# See LICENSE.TXT
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import os
+import subprocess
+from pathlib import Path
+from .base import Suite, Benchmark
+from options import options
+from utils.utils import git_clone, download, run, create_build_path
+from utils.result import Result
+
+
+class GromacsBench(Suite):
+
+    def git_url(self):
+        return "https://gitlab.com/gromacs/gromacs.git"
+
+    def git_tag(self):
+        return "v2025.1"
+
+    def grappa_url(self):
+        return "https://zenodo.org/record/11234002/files/grappa-1.5k-6.1M_rc0.9.tar.gz"
+
+    def grappa_file(self):
+        return Path(os.path.basename(self.grappa_url()))
+
+    def __init__(self, directory):
+        self.directory = Path(directory).resolve()
+        model_path = str(self.directory / self.grappa_file()).replace(".tar.gz", "")
+        self.grappa_dir = Path(model_path)
+        build_path = create_build_path(self.directory, "gromacs-build")
+        self.gromacs_build_path = Path(build_path)
+        self.gromacs_src = self.directory / "gromacs-repo"
+
+    def name(self):
+        return "Gromacs Bench"
+
+    def benchmarks(self) -> list[Benchmark]:
+        return [
+            GromacsBenchmark(self, "0006", "pme", "graphs"),
+            GromacsBenchmark(self, "0006", "pme", "eager"),
+            GromacsBenchmark(self, "0192", "rf", "graphs"),
+            GromacsBenchmark(self, "0192", "rf", "eager"),
+        ]
+
+    def setup(self):
+        self.gromacs_src = git_clone(
+            self.directory,
+            "gromacs-repo",
+            self.git_url(),
+            self.git_tag(),
+        )
+
+        # Build GROMACS
+        run(
+            [
+                "cmake",
+                f"-S {str(self.directory)}/gromacs-repo",
+                f"-B {self.gromacs_build_path}",
+                f"-DCMAKE_BUILD_TYPE=Release",
+                f"-DCMAKE_CXX_COMPILER=clang++",
+                f"-DCMAKE_C_COMPILER=clang",
+                f"-DGMX_GPU=SYCL",
+                f"-DGMX_SYCL_ENABLE_GRAPHS=ON",
+                f"-DGMX_FFT_LIBRARY=MKL",
+                f"-DGMX_BUILD_OWN_FFTW=ON",
+                f"-DGMX_GPU_FFT_LIBRARY=MKL",
+                f"-DGMX_GPU_NB_CLUSTER_SIZE=8",
+                f"-DGMX_OPENMP=OFF",
```

Comment:
For PVC we also should set `DGMX_GPU_NB_NUM_CLUSTER_PER_CELL_X=1`, see https://manual.gromacs.org/2025.1/install-guide/index.html#sycl-gpu-acceleration-for-intel-gpus.

Was that intentionally omitted because you want to reuse the same build across different Intel GPUS?

Reply:
Hmm... We don't have a way of specifying gpu in the benchmarks right now. Before it was always PVC, but now we are also testing with BMG.

We probably need a method in utils or somewhere that will autodetect gpu type. Something like:

```
enum GpuTypes {
  INTEL_PVC,
  INTEL_BMG,
  OTHER // we can extend this with nvidia/amd gpus etc
  ...
}

options {
  gpu = OTHER
}

detect_gpu() {
  if options.sycl is None:
    return OTHER
  output = run(`sycl-ls --verbose`)
  default_gpu = re.search(output, ...);
  if default_gpu.contains(Data Center GPU Max):
    return PVC
  if default_gpu.contains(BMG?)
    return BMG

  return OTHER
}

somewhere in main:
options.gpu = detect_gpu()
```

Reply:
Sounds like useful functionality to add, but probably scope creep for this PR. Could have a TODO comment about adding this in once the script can detect the device being targeted.

Reply:
> Sounds good. For now I suggest we add -DGMX_GPU_NB_NUM_CLUSTER_PER_CELL_X=1 and leave a TODO to make it conditional later.

FYI, this flag is not critical either way. It improves the performance slightly on PVC, and is at least compatible with BMG (whether it improves the performance or not is an open question).

Reply:
Another minor suggestion is to add `-DGMX_CYCLE_SUBCOUNTERS=ON` for more detailed time breakdown in the log files; won't have much direct effect, but if there are any performance anomalies, the `md.log` files might become slightly more useful (if they are preserved).